lib/web/join_tokens.go

```diff
@@ -70,6 +70,9 @@ type scriptSettings struct {
 }
 
 func (h *Handler) createTokenHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
+	// TODO: This API needs to switch to ProvisionTokenSpecV3 down the road
```

Comment:
This PR is already quite large and I'm not sure there's an immediate need for this endpoint to also move from accepting ProvisionTokenSpecV2 to ProvisionTokenSpecV3 - I'd be curious on perspectives here as to whether the HTTP API for the Proxy also needs to begin transitioning immediately.